platform/linux-generic/odp_ipsec.c

```diff
@@ -58,15 +58,64 @@ int odp_ipsec_capability(odp_ipsec_capability_t *capa)
 }
 
 int odp_ipsec_cipher_capability(odp_cipher_alg_t cipher,
-				odp_crypto_cipher_capability_t capa[], int num)
+				odp_ipsec_cipher_capability_t capa[], int num)
 {
-	return odp_crypto_cipher_capability(cipher, capa, num);
+	int caps = odp_crypto_cipher_capability(cipher, NULL, 0);
+	odp_crypto_cipher_capability_t crypto_capa[caps];
+	uint32_t req_iv_len;
+	int rc, i, out;
+
+	rc = odp_crypto_cipher_capability(cipher, crypto_capa, caps);
+	if (rc <= 0)
+		return rc;
```

Comment:
return -1 is more valid here. What is return 0 is not clear. From api it's also not really clear what 0 means. validation test has assert for > 0 and does not expect 0.

Reply:
Basically I wanted to keep return from odp_crypto_cipher_capability. `0` here would mean that there are no capabilities to report.

Reply:
Since `odp_ipsec_xxx_capability()` is just a filter on top of the `odp_crypto_xxx_capability()` base layer, passing the RC through seems reasonable here. The Crypto API will be returning RC = -1 if there was a problem anyway.